### DISCUSSION TITLE
Lesson 6 while using forge Create Command i get an Error

### DISCUSSION THREAD

**Discussion by shoaib-1337** [Cyfrin/foundry-full-course-cu#310] [Q&A] [answered]
<img width="791" alt="forge" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/125559420/20b0f2a6-2d8c-4e2e-afad-dc9fbdff06f9">
While using forge create command like this  forge create SimpleStorage --rpc-url http://127.0.0.1:7545 --interactive
i get the following error: 

[⠰] Compiling...
No files changed, compilation skipped
Error: 
could not find artifact: `SimpleStorage`

What is the fix?

**Comment by shikhar229169** [comment]
Ensure that the name of contract is ```SimpleStorage```, or else it will throw this error.

**Reply by shoaib-1337** [reply]
@shikhar229169 The name is fine it is SimpleStorage

**Reply by shikhar229169** [reply]
One thing I want to make sure is, can you just do a little change in your contract and save it then run forge compile. So I want to know what it says - no files changed or compilation successful.

**Reply by shoaib-1337** [reply]
It says : No files changed, compilation skipped

**Reply by shikhar229169** [reply]
Yes, that's the problem. Its not detecting your smart contract.
Ensure that the contracts are in src folder. 
But if they are in there, please push your code to github and provide the repo link.

**Reply by shoaib-1337** [reply]
it is in the src folder 
Here is the link to repo :
https://github.com/shoaib-1337/Foundry-SimpleStorage

**Reply by shikhar229169** [reply]
I cloned your repo and it works fine.
```bash
[⠊] Compiling...
No files changed, compilation skipped
Enter private key: 
Deployer: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
Deployed to: 0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9
Transaction hash: 0xca46fd7ea141b0c3032a2ea284534bf241bbae747a46f754bbca16fac9f11d75
```

**Reply by shikhar229169** [reply]
I can see that you don't have any lib folder in your repo.
So can you make a new folder, and give the command
```
foundryup
```
and then run:
```
forge init
```

Then delete the contents of src, test, script folder and paste your SimpleStorage.sol file in src.
And try compiling and deploying

**Reply by shoaib-1337** [reply]
Still Same error. I cannot see the lib folder in explorer but -ls command shows lib folder in terminal. 
<img width="959" alt="forge pen" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/125559420/34499e7d-7407-4108-aba6-345b1468354e">

**Reply by shikhar229169** [reply, marked as answer]
Actually see your shell, it says foundry-23/foundry-test. But you have opened a different folder, in your pane.
So just type there:
```
code .
```
And then a new vs code window will open, so you have to work at that window.

**Reply by shoaib-1337** [reply]
ahh yes, Thank you.
